### ISSUE TITLE
REST API: User with "edit product" capability only cannot create new product for submit review

### ISSUE THREAD

**Issue by whyka** [woothemes/woocommerce#10379]
I want to thank you for this great plugin.
However I have a problem with user roles. My site allows contributors (with "edit product" capability) create new product via review stage. This feature works fine on Dashboard but failed on API POST /wc-api/v3/products. Note that I already set status to draft or pending
```
{
  "product": {
    "status": "pending",
...
```

As I read the source code, the following causes the problem:
```
// Check permissions.
if ( ! current_user_can( 'publish_products' ) ) {
	throw new WC_API_Exception( 'woocommerce_api_user_cannot_create_product', __( 'You do not have permission to create products', 'woocommerce' ), 401 );
}

I think it is helpful if can extend WooCommerce but not figure out how.
Thank you for any help.

**Comment by claudiosmweb** [comment]
We'll add some filters for permissions soon.

**Comment by claudiosmweb** [comment]
Fixed in our new API #10435
